subsys/bluetooth/audio/audio_iso.c

```diff
@@ -0,0 +1,220 @@
+		bool iter;
+
+		if (!iso) {
+			continue;
+		}
+
+		iter = func(iso, user_data);
+		bt_audio_iso_unref(iso);
+
+		if (!iter) {
+			return;
+		}
+	}
+}
+
+struct bt_audio_iso_find_param {
+	struct bt_audio_iso *iso;
+	bt_audio_iso_func_t func;
+	void *user_data;
+};
+
+static bool bt_audio_iso_find_cb(struct bt_audio_iso *iso, void *user_data)
+{
+	struct bt_audio_iso_find_param *param = user_data;
+	bool found;
+
+	found = param->func(iso, param->user_data);
+	if (found) {
+		param->iso = bt_audio_iso_ref(iso);
+	}
+
+	return !found;
+}
+
+struct bt_audio_iso *bt_audio_iso_find(bt_audio_iso_func_t func,
+				       void *user_data)
+{
+	struct bt_audio_iso_find_param param = {
+		.iso = NULL,
+		.func = func,
+		.user_data = user_data,
+	};
+
+	bt_audio_iso_foreach(bt_audio_iso_find_cb, &param);
+
+	return param.iso;
+}
+
+void bt_audio_iso_init(struct bt_audio_iso *iso, struct bt_iso_chan_ops *ops)
+{
+	iso->chan.ops = ops;
+	iso->chan.qos = &iso->qos;
+
+	/* Setup points for both Tx and Rx
+	 * This is due to the limitation in the ISO API where pointers like
+	 * the `qos->tx` shall be initialized before the CIS is connected if
+	 * ever want to use it for TX, and ditto for RX. They cannot be
+	 * initialized after the CIS has been connected
+	 */
+	iso->chan.qos->rx = &iso->rx.qos;
+	iso->chan.qos->rx->path = &iso->rx.path;
+	iso->chan.qos->rx->path->cc = iso->rx.cc;
+
+	iso->chan.qos->tx = &iso->tx.qos;
+	iso->chan.qos->tx->path = &iso->tx.path;
+	iso->chan.qos->tx->path->cc = iso->tx.cc;
+}
+
+void bt_audio_iso_bind_ep(struct bt_audio_iso *iso, struct bt_audio_ep *ep)
+{
+	struct bt_iso_chan_qos *qos;
+
+	__ASSERT_NO_MSG(ep != NULL);
+	__ASSERT_NO_MSG(iso != NULL);
+	__ASSERT(ep->iso == NULL, "ep %p bound with iso %p already", ep, ep->iso);
+	__ASSERT(ep->dir == BT_AUDIO_DIR_SINK || ep->dir == BT_AUDIO_DIR_SOURCE,
+		 "invalid dir: %u", ep->dir);
+
+	qos = iso->chan.qos;
+
+	if (ep->dir == BT_AUDIO_DIR_SINK) {
+		__ASSERT(iso->rx.ep == NULL,
+			 "iso %p bound with ep %p", iso, iso->rx.ep);
+		iso->rx.ep = ep;
+		/* qos->rx already set on init */
+	} else {
+		__ASSERT(iso->tx.ep == NULL,
+			 "iso %p bound with ep %p", iso, iso->tx.ep);
+		iso->tx.ep = ep;
+		/* qos->tx already set on init */
+	}
+
+	ep->iso = bt_audio_iso_ref(iso);
+}
+
+void bt_audio_iso_unbind_ep(struct bt_audio_iso *iso, struct bt_audio_ep *ep)
+{
+	struct bt_iso_chan_qos *qos;
+
+	__ASSERT_NO_MSG(ep != NULL);
+	__ASSERT_NO_MSG(iso != NULL);
+	__ASSERT(ep->iso, "ep %p not bound with iso", iso, ep);
+	__ASSERT(ep->dir == BT_AUDIO_DIR_SINK || ep->dir == BT_AUDIO_DIR_SOURCE,
+		 "Invalid dir: %u", ep->dir);
+
+	qos = iso->chan.qos;
+
+	if (ep->dir == BT_AUDIO_DIR_SINK) {
+		__ASSERT(iso->rx.ep == ep,
+			 "iso %p not bound with ep %p", iso, ep);
+		iso->rx.ep = NULL;
+		qos->rx = NULL;
+	} else  {
+		__ASSERT(iso->tx.ep == ep,
+			 "iso %p not bound with ep %p", iso, ep);
+		iso->tx.ep = NULL;
+		qos->tx = NULL;
```

Comment:
```suggestion
		iso->rx.ep = NULL;
	} else  {
		__ASSERT(iso->tx.ep == ep,
			 "iso %p not bound with ep %p", iso, ep);
		iso->tx.ep = NULL;
```

Reply:
Hmm, when I applied the changes broadcaster tests stopped working for me

Reply:
Really? That's odd. The RX and TX pointer should be set by the `init` function, right? 

The pointers are basically constant, and are just pointing to structs within the `bt_audio_iso`, so the pointer shouldn't change at all.

Reply:
ha! found it :)